imagebuildah/stage_executor.go

```diff
@@ -1201,6 +1232,10 @@ func (s *StageExecutor) Execute(ctx context.Context, base string) (imgID string,
 				}
 			}
 		} else {
+			// This log line is majorly here so we can verify in tests
+			// that our cache is performing in the most optimal way for
+			// various cases.
+			logrus.Debugf("Found a cache hit in the first iteration with id %s", cacheID)
 			// If the instruction would affect our configuration,
```

Comment:
(Unrelated? @nalind As someone unfamiliar with the codebase, the comment about “all that we need” … is unclear to me. Didn’t we get an exact cache hit against the current step’s command, i.e. doesn’t the current `cacheID` image already contain _exactly_ the correct configuration?

It seems that we only ever use the `cacheID` image for the filesystem contents, and we _always_ build the config ourselves in `ib` (at least I can’t see anything that would update things like `ib.Author` from `cacheID`) — so is this comment basically saying “we got the filesystem we want, now update the pending config state ourselves”?

The “_still_ don’t want to restart using the image’s configuration blob” wording somewhat tantalizingly suggests to me that sometimes we _might_ want to restart using it…)

Reply:
In days when we didn't have robust storage options for unprivileged users, discarding a working container and creating a new one based on a cache hit was expensive compared to merely letting the imagebuilder library update an in-memory configuration structure and moving on.